### DISCUSSION TITLE
About requiring JDK 11

### DISCUSSION THREAD

**Discussion by arcuri82** [JSQLParser/JSqlParser#1991] [General]
Hi,
  many thanks for this project. 
From the documentation, I see that 4.9 is the last version supporting JDK 8, and development on master branch now requires JDK 11. 
I will not bore you with the details, but this is a major issue for the projects I use JSqlParser in, as most likely we will need to support JDK 8 until the heat death of the universe :(  (after all, JDK 8 is still the most used version according to [JetBrains's 2023 survey](https://www.jetbrains.com/lp/devecosystem-2023/java/)). 

If I can take a moment of your time, may I ask why are you moving to JDK 11? Is it just to use new features of Java? or are there dependencies you use in JSqlParser that jumped to JDK 11 as well? Just trying to estimate how difficult would it be to maintain a JDK 8 fork with backports of needed new features/bug-fixes from main JSqlParser repository.

Again, many thanks for this very useful project

**Comment by manticore-projects** [comment]
Greetings @arcuri82 and thank you for your inquiry.

We fully understand your concern and I have been maintaining Java 8 compatibility for a long time until the last customer also moved to Java 11 finally.

The main reason to switch to Java 8 now is the building framework. More and more plugins like CheckStyle and PMD and Spotless are depending on Java 11 already, some others start to insist in Java 17 already. IDEs insist in Java 11.

Furthermore, when writing code it becomes harder to remember, what was not available on Java 8 yet. And last but not least we should not fall too far behind the modern tool chain or else the technical liability will grow too big one day.

Last but not least I don't see the challenge yet: We don't remove the JDK 8 artifacts from any repo and JSQLParser 4.9 was pretty complete and saw a very low issue rate. You can stick with those versions for many years to come and as long as your Java 8 projects live on.

Only if you really need a particular new feature for an actively developed project you would need to upgrade to JSQLParser 4.10 plus -- but why would a "dead" Java 8 software require that?

**Comment by arcuri82** [comment]
Hi @manticore-projects ,

 Thanks for taking the time to share your thoughts on this matter.
I don't want to take more of your time, but for sake of discussion let me have 2 final comments:

1) dealing with JDK 8 doesn't mean a project is dead. In our case, we use JSQLParser in an open-source library. Although we make sure it works with latest versions of JDK, we don't feel like moving to JDK 11 yet, at least until usage statistics of JDK 8 drop significantly (like JDK 7 for example)

2) may I be so bold to suggest to please call the next version of JSQLParser 5.0 instead of 4.10? Having major breaking changes in a minor version release might be inconvenient/confusing for some users of this library 

Again, many thanks

**Comment by manticore-projects** [comment]
> * dealing with JDK 8 doesn't mean a project is dead. In our case, we use JSQLParser in an open-source library. Although we make sure it works with latest versions of JDK, we don't feel like moving to JDK 11 yet, at least until usage statistics of JDK 8 drop significantly (like JDK 7 for example)

Point well taken and you will be able to continue your project using JSQLParser 4.9 which was an ideal "breaking point" because it has been much feature complete and you won't miss out much for quite a time.
 
> * may I be so bold to suggest to please call the next version of JSQLParser 5.0 instead of 4.10? Having major breaking changes in a minor version release might be inconvenient/confusing for some users of this library

Yes, we are considering this of course. Thank you for your advice and input.
